src/main/java/org/openjdk/jextract/impl/StructBuilder.java

```diff
@@ -139,40 +139,58 @@ public void addVar(Declaration.Variable varTree) {
             String sizeField = emitSizeFieldDecl(varTree);
             emitSegmentGetter(javaName, offsetField, sizeField);
         } else if (Utils.isPointer(varTree.type()) || Utils.isPrimitive(varTree.type())) {
-            emitFieldGetter(javaName, varTree, offsetField);
-            emitFieldSetter(javaName, varTree, offsetField);
+            String layoutField = emitLayoutFieldDecl(varTree);
+            emitFieldGetter(javaName, varTree, layoutField, offsetField);
+            emitFieldSetter(javaName, varTree, layoutField, offsetField);
         } else {
             throw new IllegalArgumentException(STR."Type not supported: \{varTree.type()}");
         }
     }
 
+    private List<String> prefixNamesList() {
+        return nestedAnonDeclarations.stream()
+                .map(d -> AnonymousStruct.anonName((Declaration.Scoped)d))
+                .toList().reversed();
+    }
+
+    private String fieldElementPaths(String nativeName) {
+        StringBuilder builder = new StringBuilder();
+        String prefix = "";
+        for (String prefixElementName : prefixNamesList()) {
+            builder.append(prefix + "groupElement(\"" + prefixElementName + "\")");
+            prefix = ", ";
+        }
+        builder.append(prefix + "groupElement(\"" + nativeName + "\")");
+        return builder.toString();
+    }
+
     private void emitFieldDocComment(Declaration.Variable varTree, String header) {
         incrAlign();
         emitDocComment(varTree, header);
         decrAlign();
     }
 
-    private void emitFieldGetter(String javaName, Declaration.Variable varTree, String offsetField) {
+    private void emitFieldGetter(String javaName, Declaration.Variable varTree, String layoutField, String offsetField) {
         String seg = safeParameterName("seg");
         Class<?> type = Utils.carrierFor(varTree.type());
         appendBlankLine();
         emitFieldDocComment(varTree, "Getter for field:");
         appendIndentedLines(STR."""
             public static \{type.getSimpleName()} \{javaName}(MemorySegment \{seg}) {
-                return \{seg}.get(\{layoutString(varTree.type())}, \{offsetField});
+                return \{seg}.get(\{layoutField}, \{offsetField});
```

Comment:
overall, not too sure whether using separate layout/offset here is any better than using a var handle... opinions?